search/materialize.go

```diff
@@ -314,20 +319,24 @@ func (m *Materializer) materializeColumn(ctx context.Context, file *storage.Parq
 
 		for _, r := range rr {
 			if pageRowRange.Overlaps(r) {
-				pagesToRowsMap[i] = append(pagesToRowsMap[i], r)
+				pagesToRowsMap[i] = append(pagesToRowsMap[i], pageRowRange.Intersection(r))
```

Comment:
imagine rr contains one range [0-100] then pagesToRowsMap[anything] = [0-100] essentially, the partition will subsequently only contain [0-100] and hilarity will happen later.